File: api/src/models/drone.py
import uuid
from datetime import datetime

from sqlalchemy import String, Float, Integer, DateTime, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column


class Base(DeclarativeBase):
    pass


class Drone(Base):
    __tablename__ = "drones"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    name: Mapped[str] = mapped_column(String(255), unique=True, nullable=False)
    arm_count: Mapped[int] = mapped_column(Integer, default=4)
    mass_kg: Mapped[float] = mapped_column(Float, nullable=False)
    arm_length_m: Mapped[float] = mapped_column(Float, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )

File: frontend/src/types/drone.ts
export interface Drone {
  id: string;
  name: string;
  arm_count: number;
  mass_kg: number;
  arm_length_m: number;
  created_at: string;
  updated_at: string;
}

export interface DroneCreate {
  name: string;
  mass_kg: number;
  arm_length_m: number;
}

export interface DroneUpdate {
  name?: string;
  mass_kg?: number;
  arm_length_m?: number;
}
